### DISCUSSION TITLE
insertBefore equivalent?

### DISCUSSION THREAD

**Discussion by sevenc-nanashi** [vanjs-org/van#341] [Q&A] [answered]
I'd like to do `van.add` before a specific child, how do I do that?

**Comment by sirenkovladd** [comment, marked as answer]
[van.add](https://github.com/vanjs-org/van/blob/main/src/van.js#L85-L93) is just a call to the native method [append](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) you can use any other function instead of `van.add`.

**Reply by sevenc-nanashi** [reply]
Oh, things like states works without van.add? Awesome! Thanks.

**Reply by Tao-VanJS** [reply]
Hi @sevenc-nanashi,

You probably need some workaround to insert a state-derived DOM node. Here is the technique you can try:

```js
const counter = van.state(0)
parent.insertBefore(div(counter).firstChild, refNode)
```

**Reply by aptash** [reply]
Shouldn't we add `insertBefore` to VanJS the way `append` was added?
Meanwhile it's not clear to me - why the `append` was added but the other methods weren't.
Could you please explain?

**Reply by Tao-VanJS** [reply]
Hi @aptash,

We want to keep the public API of VanJS simple. `van.add` is suitable for most of use cases. For other less common scenarios, users can use some workaround to achieve their goals.
